powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/annotation/PowerAuthEncryptionArgumentResolver.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * PowerAuth integration libraries for RESTful API applications, examples and
+ * related software components
+ *
+ * Copyright (C) 2017 Lime - HighTech Solutions s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.getlime.security.powerauth.rest.api.spring.annotation;
+
+import io.getlime.security.powerauth.rest.api.base.encryption.PowerAuthEciesEncryption;
+import org.springframework.core.MethodParameter;
+import org.springframework.lang.NonNull;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Argument resolver for {@link PowerAuthEciesEncryption} objects. It enables automatic
+ * parameter resolution for methods that are annotated via {@link PowerAuthEciesEncryption} annotation.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public class PowerAuthEncryptionArgumentResolver implements HandlerMethodArgumentResolver {
+
+    @Override
+    public boolean supportsParameter(@NonNull MethodParameter parameter) {
```

Comment:
We need to use some annotation on an actual controller handler method parameter, or maybe a hint in the annotation to be able to determine the expected parameter type, or maybe check for the `RequestBody` and `ResponseBody` annotations. The idea is that the programmer should be able to write something like this:

```java
@RequestMapping(value="endpoint", httpMethod=POST)
@Encrypted
public @ResponseBody MyResponse process(@RequestBody MyRequest) {
   /// code
   return response;
}
```

Reply:
I implemented support for decryption and encryption using `@PowerAuthEncryption` annotation and `@EncryptedRequestBody` in Spring.

I suggest we leave the JAX-RS implementation for later (although it is much simpler), because it is currently not on critical path - I will raise a separate ticket.